src/py-opentimelineio/opentimelineio/plugins/manifest.py

```diff
@@ -250,71 +245,59 @@ def load_manifest():
 
             result.extend(manifest_from_file(json_path))
 
-    # setuptools.pkg_resources based plugins
-    if pkg_resources:
-        for plugin in pkg_resources.iter_entry_points(
-                "opentimelineio.plugins"
-        ):
-            plugin_name = plugin.name
+    try:
+        entry_points = metadata.entry_points(group='opentimelineio.plugins')
+    except TypeError:
+        # For python <= 3.9
+        entry_points = metadata.entry_points().get('opentimelineio.plugins', [])
+
+    for plugin in entry_points:
+        plugin_name = plugin.name
+        try:
+            plugin_entry_point = plugin.load()
             try:
-                plugin_entry_point = plugin.load()
-                try:
-                    plugin_manifest = plugin_entry_point.plugin_manifest()
+                plugin_manifest = plugin_entry_point.plugin_manifest()
 
-                    # this ignores what the plugin_manifest() function might
-                    # put into source_files in favor of using the path to the
-                    # python package as the unique identifier
+                # this ignores what the plugin_manifest() function might
+                # put into source_files in favor of using the path to the
+                # python package as the unique identifier
 
-                    manifest_path = os.path.abspath(
-                        plugin_entry_point.__file__
-                    )
+                manifest_path = os.path.abspath(
+                    plugin_entry_point.__file__
+                )
 
-                    if manifest_path in result.source_files:
-                        continue
+                if manifest_path in result.source_files:
+                    continue
 
-                    plugin_manifest.source_files = [manifest_path]
-                    plugin_manifest._update_plugin_source(manifest_path)
+                plugin_manifest.source_files = [manifest_path]
+                plugin_manifest._update_plugin_source(manifest_path)
 
-                except AttributeError:
-                    if not pkg_resources.resource_exists(
-                            plugin.module_name,
-                            'plugin_manifest.json'
-                    ):
-                        raise
+            except AttributeError:
+                if not resources.is_resource(plugin.value, "plugin_manifest.json"):
+                    raise
 
-                    filepath = os.path.abspath(
-                        pkg_resources.resource_filename(
-                            plugin.module_name,
-                            'plugin_manifest.json'
-                        )
-                    )
+                with resources.path(plugin.value, "plugin_manifest.json") as filepath:
```

Comment:
`path` is deprecated. [files](https://docs.python.org/3/library/importlib.resources.html#importlib.resources.files) should be used instead.

Also I believe this except block can simplified:

```
name = plugin_entry_point.__name__
filepath = importlib.resources.files(name) / "plugin_manifest.json"

if filepath.as_posix() in result.source_files:
    continue

try:
    plugin_manifest = core.deserialize_json_from_string(
        filepath.read_text()
    )
    plugin_manifest._update_plugin_source(filepath.as_posix())
    plugin_manifest.source_files.append(filepath.as_posix())
except IOError:
    raise
```

Reply:
Thanks, @JeanChristopheMorinPerso - I love it.

Updating `otio_mockplugin` and `otio_counter` to match and adding support for `3.7`.